Error: '<' not supported between instances of 'NoneType' and 'NoneType'

![image](https://user-images.githubusercontent.com/5675912/50012557-ff386800-ffcf-11e8-98ac-159fd303a642.png)

logs:
```
2018-12-14 15:41:25,983 [Analytics] [ERROR]  handle_analytic_task exception: 'Traceback (most recent call last):
  File "bin/../analytics/analytic_unit_manager.py", line 77, in handle_analytic_task
    result_payload = await self.__handle_analytic_task(task)
  File "bin/../analytics/analytic_unit_manager.py", line 65, in __handle_analytic_task
    data = prepare_data(payload['data'])
  File "bin/../analytics/analytic_unit_manager.py", line 31, in prepare_data
    if not np.isnan(min(data['value'])):
TypeError: '<' not supported between instances of 'NoneType' and 'NoneType'
```

--- a/analytics/analytics/analytic_unit_manager.py
+++ b/analytics/analytics/analytic_unit_manager.py
@@ -30,9 +30,7 @@ def prepare_data(data: list):
         - converts 'timestamp' column to pd.Datetime,
         - subtracts min value from dataset
     """
-    
     data = pd.DataFrame(data, columns=['timestamp', 'value'])
-    
     data['timestamp'] = pd.to_datetime(data['timestamp'], unit='ms')
     if not np.isnan(data['value'].min()):
         data['value'] = data['value'] - min(data['value'])

--- a/analytics/analytics/analytic_unit_manager.py
+++ b/analytics/analytics/analytic_unit_manager.py
@@ -31,7 +31,7 @@ def prepare_data(data: list):
         - subtracts min value from dataset
     """
     data = pd.DataFrame(data, columns=['timestamp', 'value'])
-
+    
     data['timestamp'] = pd.to_datetime(data['timestamp'], unit='ms')
     if not np.isnan(min(data['value'])):
         data['value'] = data['value'] - min(data['value'])

--- a/analytics/analytics/analytic_unit_manager.py
+++ b/analytics/analytics/analytic_unit_manager.py
@@ -30,6 +30,7 @@ def prepare_data(data: list):
         - converts 'timestamp' column to pd.Datetime,
         - subtracts min value from dataset
     """
+    print(type(data))
     data = pd.DataFrame(data, columns=['timestamp', 'value'])
     
     data['timestamp'] = pd.to_datetime(data['timestamp'], unit='ms')

--- a/analytics/analytics/analytic_unit_manager.py
+++ b/analytics/analytics/analytic_unit_manager.py
@@ -30,11 +30,11 @@ def prepare_data(data: list):
         - converts 'timestamp' column to pd.Datetime,
         - subtracts min value from dataset
     """
-    print(type(data))
+    
     data = pd.DataFrame(data, columns=['timestamp', 'value'])
     
     data['timestamp'] = pd.to_datetime(data['timestamp'], unit='ms')
-    if not np.isnan(min(data['value'])):
+    if not np.isnan(data['value'].min()):
         data['value'] = data['value'] - min(data['value'])
 
     return data

--- a/analytics/tests/test_dataset.py
+++ b/analytics/tests/test_dataset.py
@@ -100,6 +100,13 @@ class TestDataset(unittest.TestCase):
             model.fit(dataframe, segments, dict())
         except ValueError:
             self.fail('Model {} raised unexpectedly'.format(model_name))
+    
+    def test_prepare_data_for_nan(self):
+        data_val = [np.NaN, np.NaN, np.NaN, np.NaN, np.NaN,  np.NaN, np.NaN]
+        data_ind = [1523889000000 + i for i in range(len(data))]
+        data = {data_val, data_ind}
+        result = create_dataframe(data_val)
+        self.assertEqual(prepare_data(data), result)
 
 if __name__ == '__main__':
     unittest.main()

--- a/analytics/tests/test_dataset.py
+++ b/analytics/tests/test_dataset.py
@@ -102,13 +102,19 @@ class TestDataset(unittest.TestCase):
         except ValueError:
             self.fail('Model {} raised unexpectedly'.format(model_name))
     
+    def test_prepare_data_for_nonetype(self):
+        data = [[1523889000000, None], [1523889000001, None], [1523889000002, None]]
+        try:
+            data = prepare_data(data)
+        except ValueError:
+            self.fail('Model {} raised unexpectedly'.format(model_name))  
+    
     def test_prepare_data_for_nan(self):
-        data_val = [[np.NaN, 1523889000000], [np.NaN, 1523889000001], [np.NaN, 1523889000002], [
-            np.NaN, 1523889000003], [np.NaN, 1523889000004],  [np.NaN, 1523889000005], [np.NaN, 1523889000006]]
-        #data_ind = [1523889000000 + i for i in range(len(data_val))]
-        #data = {'value': data_val, 'index': data_ind}
-        data_val = create_dataframe(data_val)
-        self.assertEqual(prepare_data(data_val), data_val)
+        data = [[1523889000000, np.NaN], [1523889000001, np.NaN], [1523889000002, np.NaN]]
+        try:
+            data = prepare_data(data)
+        except ValueError:
+            self.fail('Model {} raised unexpectedly'.format(model_name))
 
 if __name__ == '__main__':
     unittest.main()

--- a/analytics/tests/test_dataset.py
+++ b/analytics/tests/test_dataset.py
@@ -103,7 +103,8 @@ class TestDataset(unittest.TestCase):
             self.fail('Model {} raised unexpectedly'.format(model_name))
     
     def test_prepare_data_for_nan(self):
-        data_val = [np.NaN, np.NaN, np.NaN, np.NaN, np.NaN,  np.NaN, np.NaN]
+        data_val = [[np.NaN, 1523889000000], [np.NaN, 1523889000001], [np.NaN, 1523889000002], [
+            np.NaN, 1523889000003], [np.NaN, 1523889000004],  [np.NaN, 1523889000005], [np.NaN, 1523889000006]]
         #data_ind = [1523889000000 + i for i in range(len(data_val))]
         #data = {'value': data_val, 'index': data_ind}
         data_val = create_dataframe(data_val)

--- a/analytics/tests/test_dataset.py
+++ b/analytics/tests/test_dataset.py
@@ -1,5 +1,6 @@
 import unittest
 import pandas as pd
+import numpy as np 
 from analytic_unit_manager import prepare_data
 import models
 
@@ -103,10 +104,10 @@ class TestDataset(unittest.TestCase):
     
     def test_prepare_data_for_nan(self):
         data_val = [np.NaN, np.NaN, np.NaN, np.NaN, np.NaN,  np.NaN, np.NaN]
-        data_ind = [1523889000000 + i for i in range(len(data))]
-        data = {data_val, data_ind}
-        result = create_dataframe(data_val)
-        self.assertEqual(prepare_data(data), result)
+        #data_ind = [1523889000000 + i for i in range(len(data_val))]
+        #data = {'value': data_val, 'index': data_ind}
+        data_val = create_dataframe(data_val)
+        self.assertEqual(prepare_data(data_val), data_val)
 
 if __name__ == '__main__':
     unittest.main()

